pkg/aws/helpers.go

```diff
@@ -249,6 +253,11 @@ func UserTagValidator(input interface{}) error {
 				return fmt.Errorf("invalid tag format, Tags are comma separated, for example: 'foo:bar,bar:baz'")
 			}
 			tag := strings.Split(t, ":")
+
+			if UserTagReservedKeysRE.MatchString(strings.ToLower(tag[0])) {
```

Comment:
Would it be better to use insensitive check on the regex instead by adding `(?i)` to the start of the regex?